src/llm/http_llm_calculator.cc

```diff
@@ -0,0 +1,523 @@
+        if (it != this->doc.MemberEnd()) {
+            if (!it->value.IsInt())
+                return false;
+            this->maxTokens = it->value.GetInt();
+            if (this->maxTokens.value() <= 0)
+                return false;
+        }
+
+        // presence_penalty: float; optional - defaults to 0
+        it = this->doc.FindMember("presence_penalty");
+        if (it != this->doc.MemberEnd()) {
+            if (!it->value.IsDouble())
+                return false;
+            this->presencePenalty = it->value.GetDouble();
+            if (this->presencePenalty < -2.0f || this->presencePenalty > 2.0f)
+                return false;
+        }
+
+        // temperature: float; optional - defaults to 1
+        it = this->doc.FindMember("temperature");
+        if (it != this->doc.MemberEnd()) {
+            if (!it->value.IsDouble())
+                return false;
+            this->temperature = it->value.GetDouble();
+            if (this->temperature < 0.0f || this->temperature > 2.0f)
+                return false;
+        }
+
+        // top_p: float; optional - defaults to 1
+        it = this->doc.FindMember("top_p");
+        if (it != this->doc.MemberEnd()) {
+            if (!it->value.IsDouble())
+                return false;
+            this->topP = it->value.GetDouble();
+            if (this->topP < 0.0f || this->topP > 1.0f)
+                return false;
+        }
+
+        // logit_bias TODO
+        // logprops TODO
+        // top_logprobs TODO
+        // n TODO
+        // response_format TODO
+        // seed TODO
+        // stop TODO
+        // stream_options TODO
+        // tools TODO
+        // tool_choice TODO
+        // user TODO
+        // function_call TODO (deprecated)
+        // functions TODO (deprecated)
+        return true;
+    }
+};
+
+// TODO: To be moved to CB library.
+class TextStreamer {
+    std::shared_ptr<Tokenizer> tokenizer;
+    std::vector<int64_t> tokenCache;
+    size_t printLen;
+public:
+    TextStreamer(std::shared_ptr<Tokenizer> tokenizer) : tokenizer(tokenizer) {}
+
+    std::optional<std::string> put(int64_t token) {
+        tokenCache.push_back(token);
+        std::string text = tokenizer->decode(tokenCache);
+
+        if (!text.empty() && '\n' == text.back()) {
+            std::string chunk = std::string{text.data() + printLen, text.size() - printLen};
+            tokenCache.clear();
+            printLen = 0;
+            return chunk;
+        }
+        else if (text.size() >= 3 && text.compare(text.size() - 3, 3, "�") == 0) {
+            return std::nullopt;
+        } else {
+            std::string chunk = std::string{text.data() + printLen, text.size() - printLen};
+            printLen = text.size();
+            return chunk;
+        }
+        return std::nullopt;
+    }
+};
+
+using InputDataType = ovms::HttpPayload;
+using OutputDataType = std::string;
+
+const std::string LLM_SESSION_SIDE_PACKET_TAG = "LLM_NODE_RESOURCES";
```

Comment:
Wonder if we could import that from another file. I think node resources or mediapipe graph definition also use that name.

Reply:
It is only used in 2 places - 2 calculators (grpc and http), I really do not want to create another header for that and use global var